### PR TITLE
ADD: Detect device theme preference

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bkjw3gn07zdry6jx339ffx709g

[download apk](http://filestorage.bluewallet.io:1488/1593136576.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/azf916ucyvx7ft75fxrgmuc0tc

[download apk](http://filestorage.bluewallet.io:1488/1593173722.apk)

File: class/themes.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    lnbackgroundColor: '#FFFAEF',
+  },
+};
+
+export const BlueDarkTheme = {
+  dark: DarkTheme.dark,
+  colors: {...BlueDefaultTheme.colors,  ...DarkTheme.colors,
+
+    foregroundColor: '#ffffff',
+},
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply class/themes.js:39]: It's neither a component nor a class. It's a const. Still want it in 'components/' ?

**Comment by marcosrdz** [comment]
> ```
>     ✕ can import BIP84 mnemonic, fetch balance & transactions, then create a transaction (33664ms)
>     ✕ can import zpub as watch-only and create PSBT (34883ms)
> ```
> 
> e2e consistently fail

Will look into it.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/50eqqfnj9cv26wccve4d81dmdg

[download apk](http://filestorage.bluewallet.io:1488/1593286322.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/jmp56zrwe98yjked5newqe16tc

[download apk](http://filestorage.bluewallet.io:1488/1593299337.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/azppbzc4r9qrenargbaa66ztm4

[download apk](http://filestorage.bluewallet.io:1488/1593302819.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ktm6fmr3u9dy0qyvyuk1cbnd80

[download apk](http://filestorage.bluewallet.io:1488/1593320436.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/cwr1na69666pdqvky4e6z8m56r

[download apk](http://filestorage.bluewallet.io:1488/1593321365.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/x8f3aj7tv7mj2d6aumthnxja50

[download apk](http://filestorage.bluewallet.io:1488/1593363259.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2cpcem3ncba2q0eb2hr5k9y5jw

[download apk](http://filestorage.bluewallet.io:1488/1593394662.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4z427xgjed1ehy3j0yhr4nkk2m

[download apk](http://filestorage.bluewallet.io:1488/1593396819.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4g9kag4akuf9p6z3tdu0n5y5yg

[download apk](http://filestorage.bluewallet.io:1488/1593418928.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/fu754tknpmkq0vmcjw064q75jm

[download apk](http://filestorage.bluewallet.io:1488/1593419589.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8py9eg4r1abdry2a1kwrnv36eg

[download apk](http://filestorage.bluewallet.io:1488/1593420093.apk)

**Comment by Overtorment** [comment]
selftest fails

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/tef4eq8mn549ajy8wf1cta9xfm

[download apk](http://filestorage.bluewallet.io:1488/1593431480.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ndmmdbffrm6y1tee22n80g5134

[download apk](http://filestorage.bluewallet.io:1488/1593462006.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3bza1u4gv3yzbafv6b09n50yjm

[download apk](http://filestorage.bluewallet.io:1488/1593522955.apk)

File: package-lock.json [hunk 2]
```diff
@@ -5902,11 +4019,11 @@
       "integrity": "sha512-mLQ4i2QO1ytvGWFWmcngKO//JXAQueZvwEKtjgQFM4jIK0kU+ytMfplL8j+n5mspOfjHwoAg+9yhb7BwAHm36g=="
     },
     "bc-bech32": {
-      "version": "file:blue_modules/bc-bech32",
+      "version": "file:https:/registry.npmjs.org/bc-bech32/-/bc-bech32-1.0.2.tgz",
```
Comment (Overtorment) [inline package-lock.json:4022]: nein

File: screen/settings/GeneralSettings.js [hunk 3]
```diff
@@ -37,60 +37,65 @@ const GeneralSettings = () => {
...
+      ) : null}
+      <BlueListItemHooks
+        Component={TouchableWithoutFeedback}
+        title={loc.settings.enable_advanced_mode}
+        switch={{ onValueChange: onAdvancedModeSwitch, value: isAdancedModeEnabled }}
+      />
+      <BlueCard>
+        <BlueTextHooks>
+          When enabled, you will see advanced options such as different wallet types and the ability to specify the LNDHub instance you wish
+          to connect to.
```
Comment (Overtorment) [inline screen/settings/GeneralSettings.js:88]: youre reverting recent changes

File: themes.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { DefaultTheme, DarkTheme } from '@react-navigation/native';
```
Comment (Overtorment) [inline themes.js:1]: lets move it under `components/` folder. lets try to make root folder more tidy

File: themes.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    outgoingForegroundColor: '#d0021b',
+    successColor: '#37c0a1',
+    failedColor: '#ff0000',
+    shadowColor: '#000000',
+    inverseForegroundColor: '#ffffff',
+    hdborderColor: '#68BBE1',
+    hdbackgroundColor: '#ECF9FF',
+    lnborderColor: '#F7C056',
+    lnbackgroundColor: '#FFFAEF',
+    background: '#FFFFFF',
```
Comment (Overtorment) [inline themes.js:39]: if we have all colors in theme file lets remove them from `class/app-storage.js`

File: screen/settings/GeneralSettings.js (hunk 3 above)
Comment (marcosrdz) [reply screen/settings/GeneralSettings.js:88]: How so? I don't see it,.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/h96knyp1bj79yhn3gejbkyy2nc

[download apk](http://filestorage.bluewallet.io:1488/1593551264.apk)

File: screen/settings/GeneralSettings.js (hunk 3 above)
Comment (Overtorment) [reply screen/settings/GeneralSettings.js:88]: image

**Pull request closed without merging**